core/src/main/java/com/digitalpebble/stormcrawler/protocol/AbstractHttpProtocol.java

```diff
@@ -65,6 +76,134 @@ public void configure(Config conf) {
+                    LOG.error(
+                            "invalid proxy rotation scheme passed `{}` defaulting to ROUND_ROBIN; options: {}",
+                            proxyRot, ProxyRotation.values()
+                    );
+                proxyRotationScheme = ProxyRotation.ROUND_ROBIN;
+                break;
+        }
+
+        String proxyFile = ConfUtils.getString(conf, "http.proxy.file", null);
+        String proxyHost = ConfUtils.getString(conf, "http.proxy.host", null);
+
+        // conditionally create proxy manager proxy
+        if (
+                (proxyFile != null && !proxyFile.isEmpty()) ||
+                        (proxyHost != null && !proxyHost.isEmpty())
+        ) {
+            String proxyManagerImplementation = ConfUtils.getString(
+                    conf,
+                    "http.proxy.manager",
+                    // default to the multi proxy manager if a list or file was passed otherwise single
+                    (proxyFile != null && !proxyFile.isEmpty()) ?
+                            "com.digitalpebble.stormcrawler.proxy.MultiProxyManager" :
+                            "com.digitalpebble.stormcrawler.proxy.SingleProxyManager"
+            );
+
+            // create class to hold the proxy manager class loaded from the config
+            Class proxyManagerClass;
+            try {
+                proxyManagerClass = Class.forName(proxyManagerImplementation);
+                boolean interfaceOK = ProxyManager.class
+                        .isAssignableFrom(proxyManagerClass);
+                if (!interfaceOK) {
+                    throw new RuntimeException("Class "
+                            + proxyManagerImplementation
+                            + " does not implement ProxyManager");
+                }
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException("Can't load class "
+                        + proxyManagerImplementation);
+            }
+
+            LOG.info("loaded proxy manager class: {}", proxyManagerClass.getName());
+
+            // create boolean to track if proxy setup succeeded
+            boolean success = true;
+
+            // map proxy type to the proper manager
+            if (proxyFile != null && !proxyFile.isEmpty()) {
+                try {
+                    // create new proxy manager from file
+                    proxyManager = (ProxyManager) proxyManagerClass.newInstance();
+                    proxyManager.configure(proxyRotationScheme, proxyFile);
+                } catch (FileNotFoundException | InstantiationException | IllegalAccessException e) {
+                    LOG.error("failed to create proxy manager `" + proxyFile + "`", e);
+                    success = false;
+                }
+            } else {
+                // load remaining values for single proxy
+                String proxyType = ConfUtils.getString(conf, "http.proxy.type", "HTTP");
```

Comment:
same as above - these should be within the conf method of the SingleProxyManager